invenio/ext/jsonalchemy/registry.py

```diff
@@ -20,13 +20,30 @@
 from flask.ext.registry import PkgResourcesDirDiscoveryRegistry, \
     ModuleAutoDiscoveryRegistry, RegistryProxy
 
+from invenio.utils.datastructures import LaziestDict
 from invenio.ext.registry import ModuleAutoDiscoverySubRegistry
 from invenio.utils.datastructures import LazyDict
 
+from jsonalchemy.registry import MetaData
+
+
 jsonext = lambda namespace: RegistryProxy(
     namespace, ModuleAutoDiscoveryRegistry, namespace)
 
 
+def get_metadata(namespace):
+    """Return metadata object.
+
+    :param namespace: namespace name
+    :return: JSONAlchemy MetaData object
+    """
+    return MetaData(packages=jsonext(namespace))
+
+
+metadata = LaziestDict(get_metadata)
+"""Lazy metadata cache."""
+
+
 fields_proxy = lambda namespace: RegistryProxy(
```

Comment:
the `functions()` is used in `invenio/ext/elasticsearch/jsonext/parsers/elasticsearch_parser.py`.